### PR TITLE
ARROW-5208: [Python] Add mask argument to pyarrow.infer_type, do not look at masked values when inferring output type in pyarrow.array

### PR DESCRIPTION
**Pull request by wesm**

### CODE REVIEW

File: python/pyarrow/tests/test_array.py [hunk 1]
```diff
@@ -1430,6 +1430,28 @@ def test_numpy_string_overflow_to_chunked():
...
+    # length 0
+    assert pa.infer_type([], mask=[]) == pa.null()
+
+
+def test_array_masked():
+    # ARROW-5208
+    arr = pa.array([4, None, 4, 3.],
+                   mask=np.array([False, True, False, True]))
+    assert arr.type == pa.int64()
+
```
Comment (artemru) [inline python/pyarrow/tests/test_array.py]: could you also add `np.ndarray` class case :
```python
    arr = pa.array(np.array([4, None, 4, 3.], dtype="O"), mask=np.array([False, True, False, True]))
    assert arr.type == pa.int64()
```
?

Comment (wesm) [reply python/pyarrow/tests/test_array.py]: done

**Comment by wesm** [comment]
+1. Will merge when builds run

**Comment by wesm** [comment]
Appveyor looking ok here: https://ci.appveyor.com/project/wesm/arrow/builds/25511377. Merging

**Pull request closed without merging**